### PR TITLE
Bump scalafmt version to 0.4.1

### PR DESCRIPTION
**Pull request by densh**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -27,11 +25,10 @@ lazy val publishSettings = Seq(
...
       Some("releases" at nexus + "service/local/staging/deploy/maven2")
   },
-
   publishSnapshot := Def.taskDyn {
-    val travis   = Try(sys.env("TRAVIS")).getOrElse("false") == "true"
-    val pr       = Try(sys.env("TRAVIS_PULL_REQUEST")).getOrElse("false") != "false"
-    val branch   = Try(sys.env("TRAVIS_BRANCH")).getOrElse("")
+    val travis = Try(sys.env("TRAVIS")).getOrElse("false") == "true"
+    val pr = Try(sys.env("TRAVIS_PULL_REQUEST")).getOrElse("false") != "false"
+    val branch = Try(sys.env("TRAVIS_BRANCH")).getOrElse("")
```
Comment (densh) [inline build.sbt]: @olafurpg Why no alignment here?

File: build.sbt [hunk 2]
```diff
@@ -40,25 +37,26 @@ lazy val publishSettings = Seq(
...
   }.value,
-
   credentials ++= {
     for {
-      realm    <- sys.env.get("MAVEN_REALM")
-      domain   <- sys.env.get("MAVEN_DOMAIN")
-      user     <- sys.env.get("MAVEN_USER")
+      realm <- sys.env.get("MAVEN_REALM")
+      domain <- sys.env.get("MAVEN_DOMAIN")
+      user <- sys.env.get("MAVEN_USER")
```
Comment (densh) [inline build.sbt]: @olafurpg Why no alignment here?

File: build.sbt [hunk 3]
```diff
@@ -157,49 +157,53 @@ lazy val nscplugin =
...
-                                       publishLocal in javalib,
-                                       publishLocal in scalalib)
-    ).
-    dependsOn(tools)
+      publishLocal in nir,
+      publishLocal in tools,
+      publishLocal in nscplugin,
+      publishLocal in nativelib,
+      publishLocal in javalib,
+      publishLocal in scalalib)
```
Comment (densh) [inline build.sbt]: @olafurpg Indentation seems to be completely off.

**Comment by densh** [comment]
Apart from minor quirks with `build.sbt` formatting, this looks like a solid upgrade, thanks @olafurpg!

File: nir/src/main/scala/scala/scalanative/nir/Focus.scala [hunk 4]
```diff
@@ -89,11 +88,11 @@ final case class Focus(
...
     val caseblockss = cases.map(_._3)
     val blocks = finish(
-        Cf.Switch(scrut,
-                  Next.Label(defaultname, Seq()),
-                  casevals.zip(casenames).map {
-                case (v, n) => Next.Case(v, n)
-              })).blocks
+      Cf.Switch(scrut,
+                Next.Label(defaultname, Seq()),
+                casevals.zip(casenames).map {
```
Comment (olafurpg) [inline nir/src/main/scala/scala/scalanative/nir/Focus.scala]: Yay, happy to see this fix is the real world

File: sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala [hunk 5]
```diff
@@ -120,12 +112,12 @@ object ScalaNativePluginInternal {
       val dotpath       = nativeEmitDependencyGraphPath.value
       val linkage       = nativeLibraryLinkage.value
       val sharedLibrary = nativeSharedLibrary.value
-      val opts          = new NativeOpts(classpath,
-                                         abs(appll),
-                                         dotpath.map(abs),
-                                         entry,
-                                         verbose,
-                                         sharedLibrary)
+      val opts = new NativeOpts(classpath,
```
Comment (olafurpg) [inline sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Cool, this was fixed a while ago.

File: scripted-tests/run/C_argc_argv-to-java_args/project/scala-native.sbt [hunk 6]
```diff
@@ -1,7 +1,8 @@
 {
   val pluginVersion = System.getProperty("plugin.version")
-  if(pluginVersion == null)
-    throw new RuntimeException("""|The system property 'plugin.version' is not defined.
+  if (pluginVersion == null)
+    throw new RuntimeException(
+      """|The system property 'plugin.version' is not defined.
```
Comment (olafurpg) [inline scripted-tests/run/C_argc_argv-to-java_args/project/scala-native.sbt]: `assumeStandardLibraryStripMargin = true` will fix this, but it seems to be non-idempotent in some corner cases.

**Comment by densh** [comment]
Currently blocked by https://github.com/olafurpg/scalafmt/issues/481, there seems to be no workaround.
